src/code/z_demo.c

```diff
@@ -1232,684 +1404,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
                 } else {
-                    play->envCtx.screenFillColor[3] = 255.0f * temp;
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
                 }
                 break;
 
-            case 12:
-                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * temp);
+            case CS_TRANS_BLACK_HALF_FILL:
+                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * lerp);
                 break;
 
-            case 13:
+            case CS_TRANS_BLACK_HALF_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - temp) * 155.0f);
+                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - lerp) * 155.0f);
                 break;
         }
     }
 }
 
-// Command 0x1 & 0x5: Camera Eye Points
-s32 Cutscene_Command_CameraEyePoints(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 relativeToLink) {
-    s32 shouldContinue = 1;
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_UpdateCamEyeSpline(PlayState* play, CutsceneContext* csCtx, u8* script, u8 relativeToPlayer) {
+    s32 shouldContinue = true;
+    CsCmdGeneric* cmd = (CsCmdGeneric*)script;
     s32 size;
 
-    cmd += 8;
-    size = 8;
+    script += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((cmd->startFrame > csCtx->camEyeSplinePointsAppliedFrame) ||
+         (csCtx->camEyeSplinePointsAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)script;
+
+        if (csCtx->camAtReady) {
+            csCtx->camEyeSplinePointsAppliedFrame = cmd->startFrame;
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((csCtx->unk_18 < cmdBase->startFrame) || (csCtx->unk_18 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            csCtx->unk_18 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+            if (gUseCutsceneCam) {
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_CS_0);
                 Play_ChangeCameraStatus(play, sReturnToCamId, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Camera_ResetAnim(Play_GetCamera(play, csCtx->subCamId));
-                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->subCamLookAtPoints,
-                                   csCtx->subCamEyePoints, GET_PLAYER(play), relativeToLink);
+                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camAtPoints, csCtx->camEyePoints,
```

Comment:
is there any action you want taken due to these points?
the single point variations are entirely unused and because of this, this scenario was not even considered

Reply:
I mean I've suggested before to just not include a macro for the cs_cam_eye/at half-baked thing

Probably at least a mention somewhere camAtReady/camEyeReady are vaguely jank, or on the cs_cam_eye/at macros idk

Reply:
ill add 
`This feature is not used in the final game, it is recommended to use splines in all cases`

but yea i dont think the macro should just be removed

Reply:
`This feature is not used in the final game and comes with some jank, `

Reply:
if you can come up with a more professional way to say that then sure lol

Reply:
`This feature is not used in the final game and lacks polish, it is recommended to use splines in all cases`